src/validation/forks.cpp

```diff
@@ -92,41 +82,57 @@ bool IsDAAEnabled(const Consensus::Params &consensusparams, const CBlockIndex *p
     return IsDAAEnabled(consensusparams, pindexTip->nHeight);
 }
 
-bool IsNov152018Scheduled() { return miningForkTime.Value() != 0; }
-bool IsNov152018Enabled(const Consensus::Params &consensusparams, const CBlockIndex *pindexTip)
+bool IsNov152018Activated(const Consensus::Params &consensusparams, const int32_t nHeight)
+{
+    return nHeight >= consensusparams.nov2018Height;
+}
+
+bool IsNov152018Activated(const Consensus::Params &consensusparams, const CBlockIndex *pindexTip)
 {
     if (pindexTip == nullptr)
     {
         return false;
     }
-    return pindexTip->IsforkActiveOnNextBlock(miningForkTime.Value());
+    return IsNov152018Activated(consensusparams, pindexTip->nHeight);
+}
+
+bool AreWeOnBCHChain()
+{
+    const Consensus::Params &consensusparams = Params().GetConsensus();
+    return ((miningForkTime.Value() != 0) && (IsNov152018Activated(consensusparams, chainActive.Tip())));
 }
 
-bool IsNov152018Next(const Consensus::Params &consensusparams, const CBlockIndex *pindexTip)
+bool IsMay152019Enabled(const Consensus::Params &consensusparams, const CBlockIndex *pindexTip)
 {
     if (pindexTip == nullptr)
     {
         return false;
     }
-    return pindexTip->forkAtNextBlock(miningForkTime.Value());
+    return pindexTip->IsforkActiveOnNextBlock(miningForkTime.Value());
 }
 
-
-bool IsSv2018Scheduled() { return miningSvForkTime.Value() != 0; }
-bool IsSv2018Enabled(const Consensus::Params &consensusparams, const CBlockIndex *pindexTip)
+bool IsMay152019Next(const Consensus::Params &consensusparams, const CBlockIndex *pindexTip)
```

Comment:
Nit: There is only one fork in May, so maybe remove `15` for readability

Reply:
good idea. 

I was just following the naming convention we used so far. 

no problem in changing it. 

let's see what other thinks.

Reply:
@ptschip @gandrewstone what's your take? 

do you prefer `IsMay152019...` or `IsMay2019...`? 

@dagur have indeed a point the 15th is redundant info, It's not that we are going to have multiple upgrades in May (hopefully :smile:)

Reply:
makes sense

Reply:
changed in 66e065ca